timeseries/tests/unittests/models/test_models.py

```diff
@@ -26,649 +23,479 @@
-    model_paths = []
-    for model_class, prediction_length in itertools.product(TESTABLE_MODELS, TESTABLE_PREDICTION_LENGTHS):
-        temp_model_path = tempfile.mkdtemp()
+class TestAllModelsInitialization:
+    EXPECTED_MODEL_TAGS = [
+        "allow_nan",
+        "can_refit_full",
+        "can_use_train_data",
+        "can_use_val_data",
+        # Tabular tags - not used by time series models
+        "valid_oof",
+        "handles_text",
+        "can_estimate_memory_usage_static",
+    ]
+
+    def test_models_can_be_initialized(self, model_class, temp_model_path):
+        model = model_class(path=temp_model_path, freq="h", prediction_length=24)
+        assert isinstance(model, AbstractTimeSeriesModel)
+
+    def test_when_model_created_then_model_has_all_required_tags(self, model_class, temp_model_path):
+        model = model_class(path=temp_model_path)
+        model_tags = model._get_tags()
+        for tag in self.EXPECTED_MODEL_TAGS:
+            assert tag in model_tags
+        assert len(model_tags) == len(self.EXPECTED_MODEL_TAGS)
+
+
+class TestAllModelsPostTraining:
+    @pytest.fixture(scope="class", params=TESTABLE_PREDICTION_LENGTHS)
+    def trained_model(self, request, model_class, tmp_path_factory):
         model = model_class(
-            path=temp_model_path,
+            path=str(tmp_path_factory.mktemp(str(uuid.uuid4())[:6])),
             freq="h",
-            prediction_length=prediction_length,
-            hyperparameters=dummy_hyperparameters,
+            prediction_length=request.param,
         )
 
         model.fit(train_data=DUMMY_TS_DATAFRAME)
         model.score_and_cache_oof(DUMMY_TS_DATAFRAME, store_val_score=True, store_predict_time=True)
-        models[(prediction_length, repr(model_class))] = model
-        model_paths.append(temp_model_path)
-
-    yield models
-
-    for td in model_paths:
-        shutil.rmtree(td)
-
-
-@pytest.mark.parametrize("model_class", TESTABLE_MODELS)
-def test_models_can_be_initialized(model_class, temp_model_path):
-    model = model_class(path=temp_model_path, freq="h", prediction_length=24)
-    assert isinstance(model, AbstractTimeSeriesModel)
-
 
-@pytest.mark.parametrize("model_class", TESTABLE_MODELS)
-@pytest.mark.parametrize("prediction_length", [1, 5])
-@pytest.mark.parametrize("metric", AVAILABLE_METRICS)
-def test_when_fit_called_then_models_train_and_all_scores_can_be_computed(
-    model_class, prediction_length, metric, trained_models
-):
-    model = trained_models[(prediction_length, repr(model_class))]
-    score = model.score(DUMMY_TS_DATAFRAME, metric)
+        yield model
 
-    assert isinstance(score, float)
+    @pytest.mark.parametrize("metric", AVAILABLE_METRICS)
+    def test_when_score_called_then_scores_can_be_computed(self, trained_model, metric):
+        score = trained_model.score(DUMMY_TS_DATAFRAME, metric)
+        assert isinstance(score, float)
 
+    def test_when_val_score_accessed_then_value_is_set(self, trained_model):
```

Comment:
Nit: This reads like "value is set because we access the val_score". What about something like `test_when_model_is_fit_then_val_score_is_set`?

Similarly for `test_when_predict_time_accessed_then_value_is_set`.

Reply:
changed to `value_is_returned`. 

This is a confusion I see we have in the tests often, of which I'm clearly guilty as well. Given-when-then is the build-operate-check steps of the test.

- Given refers to the dependency of the test, often the injected dependency. So it's "given a model that is fit,"
- When refers to what you jimmy in the test
- Then is the asserted criterion.

Here given is ommitted because it is rather clear from the test class now 👍